### PR TITLE
connectOption

### PR DESCRIPTION
**Pull request by raymondng**
code + uri, as discussed. email also added.
@andrewconner please take a look. don't merge yet.

### CODE REVIEW

**Comment by andrewconner** [comment]
Thanks, I think this will work. Only unsure about having to redirect the user to a POST, but I think I know how.

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/core/AuthController.scala [hunk 1]
```diff
@@ -252,6 +256,18 @@ class AuthController @Inject() (
     }
   }
 
+  def connectOption() = MaybeUserAction(parse.tolerantJson) { implicit request =>
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/core/AuthController.scala]: Can the body be form URL encoded? I'm sending the user here, not using a json API here.

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/core/AuthController.scala [hunk 2]
```diff
@@ -41,6 +41,10 @@ object AuthController {
   def obscureEmailAddress(address: String) = obscureRegex.replaceFirstIn(address, """$1...@""")
 }
 
+@json case class ConnectOptionInfo(private val email: String, provider: String) {
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/core/AuthController.scala]: `provider` is formatted how? `facebook` and `email`?

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/core/AuthController.scala (hunk 1 above)
Comment (raymondng) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/core/AuthController.scala]: Ah good point. Let's do that.

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/core/AuthController.scala (hunk 2 above)
Comment (raymondng) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/core/AuthController.scala]: It's used to construct a `SocialNetworkType` so I'd say yes -- definitely with `facebook`. I am not too familiar with the `email` type but based on limited testing looks like it should work.

Comment (raymondng) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/core/AuthController.scala]: Actually given we only support FB in this flow, and we trigger it via social -- `facebook` looks like the only legitimate option here for now until we support LNKD or other providers. WDYT?

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/core/AuthController.scala [hunk 3]
```diff
@@ -252,6 +256,26 @@ class AuthController @Inject() (
...
 
+  val connectOptionForm = Form[ConnectOptionInfo](
+    mapping(
+      "email" -> EmailAddress.formMapping,
+      "provider" -> text
+    )(ConnectOptionInfo.apply)(ConnectOptionInfo.unapply)
+  )
+  def connectOption() = MaybeUserAction { implicit request =>
+    connectOptionForm.bindFromRequest.fold(
+      hasErrors = formWithErrors => BadRequest(Json.obj("error" -> formWithErrors.errors.head.message)),
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/core/AuthController.scala:267]: Returning JSON to a user-facing request?

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/core/AuthController.scala [hunk 4]
```diff
@@ -252,6 +256,26 @@ class AuthController @Inject() (
...
+      success = {
+        case info: ConnectOptionInfo =>
+          if (authHelper.emailAddressMatchesSomeKifiUser(info.emailAddress)) { // sanity check
+            Ok(authHelper.connectOptionView(info.emailAddress, info.provider))
+          } else {
+            BadRequest(Json.obj("error" -> "not_found")) // shouldn't happen
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/core/AuthController.scala:273]: Same

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/core/AuthController.scala (hunk 3 above)
Comment (raymondng) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/core/AuthController.scala:267]: Yes I was going to ask you about this. That does look like a common pattern for form-handling (see `AuthHelper`) -- I thought maybe we have a way to handle this, but probably assumed too much.

Comment (raymondng) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/core/AuthController.scala:267]: Looks like we need another endpoint or another way to handle this edge case. Will look into it.

**Pull request closed without merging**